### PR TITLE
browser(firefox): fix request frame attribution

### PR DESCRIPTION
**Pull request by JoelEinbinder**
Firefox will sometimes send multiple requests with the same http channel id. When a frame is loaded, the favicon is requested in the parent frame, but with the same channel id. This can cause the document request to report the wrong frame, causing the test 'should capture iframe navigation request' to fail. It fails consistently on my computer.

This patch adds the content policy type into the http channelId to better distinguish requests. Maybe there is something better we can do? It looks like we use channelId has request ids, so there might be more bugs with these favicon requests in playwright?

### CODE REVIEW

File: browser_patches/firefox/juggler/content/NetworkMonitor.js [hunk 1]
```diff
@@ -33,16 +33,18 @@ class NetworkMonitor {
       const frame = this._frameTree.frameForDocShell(window.docShell);
       if (!frame)
         return;
-      this._requestDetails.set(httpChannel.channelId, {
+      const typeId = httpChannel.loadInfo ? httpChannel.loadInfo.internalContentPolicyType : 1;
```
Comment (yury-s) [inline browser_patches/firefox/juggler/content/NetworkMonitor.js:36]: ... : Ci.nsIContentPolicy.TYPE_OTHER; to match the other place ?

File: browser_patches/firefox/juggler/content/NetworkMonitor.js [hunk 2]
```diff
@@ -33,16 +33,18 @@ class NetworkMonitor {
...
+      const channelKey = httpChannel.channelId + ':' + typeId;
```
Comment (yury-s) [inline browser_patches/firefox/juggler/content/NetworkMonitor.js:37]: Please add a comment explaining why we do this.

**Review by yury-s — APPROVED** [review]

**Review by dgozman — APPROVED** [review]

File: browser_patches/firefox/juggler/content/NetworkMonitor.js (hunk 2 above)
Comment (JoelEinbinder) [reply browser_patches/firefox/juggler/content/NetworkMonitor.js:40]: Done.

File: browser_patches/firefox/juggler/content/NetworkMonitor.js (hunk 1 above)
Comment (JoelEinbinder) [reply browser_patches/firefox/juggler/content/NetworkMonitor.js]: Done.

**Pull request merged**